src/context-logger.ts

```diff
@@ -53,31 +68,61 @@ export class ContextLogger {
   error(message: string, bindings: Bindings): void;
   error(message: string, error: Error, bindings: Bindings): void;
   error(message: string, errorOrBindings?: Error | Bindings, bindings?: Bindings): void {
-    const context: any = {};
+    let error;
+    const adaptedBindings = {};
     if (errorOrBindings instanceof Error) {
-      context.err = errorOrBindings;
+      error = errorOrBindings;
       if (bindings) {
-        Object.assign(context, bindings);
+        Object.assign(adaptedBindings, bindings);
       }
       // Bootstrapping logs
     } else if (typeof errorOrBindings === 'string') {
-      context.err = errorOrBindings;
+      error = errorOrBindings;
     } else if (errorOrBindings) {
-      Object.assign(context, errorOrBindings);
+      Object.assign(adaptedBindings, errorOrBindings);
     }
-    this.callInternalLogger('error', message, context);
+    this.callInternalLogger('error', message, adaptedBindings, error);
   }
 
-  private callInternalLogger(level: string, message: string, bindings: Bindings) {
+  private callInternalLogger(level: string, message: string, bindings: Bindings, error?: Error | string) {
     let logObject: Record<string, any>;
     if (typeof bindings === 'string') {
       // Bootstrapping logs
       logObject = { component: bindings };
     } else {
-      // Normal request log
-      logObject = { ...bindings, ...ContextStore.getContext() };
+      logObject = this.createLogEntry(message, bindings, error);
     }
     const logger = ContextLogger.internalLogger ?? this.fallbackLogger;
     return logger[level](logObject, ...[message, this.moduleName]);
   }
-}
\ No newline at end of file
+
+  private createLogEntry(
+    message: string,
+    bindings?: Record<string, any>,
+    error?: Error | string,
+  ): LogEntry {
+    const storeContext = ContextStore.getContext();
+    const adaptedContext = this.options.contextAdapter 
+      ? this.options.contextAdapter(storeContext)
+      : storeContext;
+
+    const { enabled = true, bindingsKey = 'bindings', contextKey = 'context' } = 
```

Comment:
```suggestion
    const { groupFieldsEnabled = true, bindingsKey = 'bindings', contextKey = 'context' } = 
```
Internally I would call this `groupFieldsEnabled` because a simple `enabled` in the code can be confusing in my opinion

Reply:
if we change the name here, the override coming form `this.options.groupFields` won't work (unless we also change the name there)

Reply:
Sorry I've got it wrong, it should be like this
```suggestion
    const { enabled: groupFieldsEnabled = true, bindingsKey = 'bindings', contextKey = 'context' } = 
```